src/parser/jobs/brd/modules/DriftingEa.tsx

```diff
@@ -0,0 +1,182 @@
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import Downtime from 'parser/core/modules/Downtime'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React, {Fragment} from 'react'
+import {Button, Message, Table} from 'semantic-ui-react'
+
+// Buffer (ms) to forgive insignificant drift
+const DRIFT_BUFFER = 2000
+
+// Buffer (ms) to allow for reopeners after a downtime
+const REOPENER_BUFFER = 4000
+const MIN_REOPENER_DOWNTIME = 15000
+
+// Timeline padding to see the drifted oGCD when you jump to the window
+const TIMELINE_PADDING = 2500
+
+interface DriftWindow {
+	start: number
+	casts: Array<Events['action']>
+}
+
+interface ConfirmedDriftWindow extends DriftWindow {
+	driftedActionId: number
+	drift: number
+	end: number
+}
+
+export class DriftingEa extends Analyser {
+	static override handle = 'drifting-ea'
+	static override title = t('brd.drifting-ea.title')`Empyreal Arrow Drift`
+
+	@dependency private data!: Data
+	@dependency private downtime!: Downtime
+	@dependency private timeline!: Timeline
+
+	private driftedWindows: ConfirmedDriftWindow[] = []
+
+	private currentWindow: DriftWindow =  {
+		start: this.parser.pull.timestamp,
+		casts: [],
+	}
+
+	override initialise() {
+		const eaId = this.data.actions.EMPYREAL_ARROW.id
+
+		const castFilter = filter<Event>()
+			.type('action')
+			.source(this.parser.actor.id)
+
+		this.addEventHook(castFilter.action(eaId), this.onDriftableCast)
+		this.addEventHook(castFilter, this.onCast)
+	}
+
+	private onDriftableCast(event: Events['action']) {
+		const action = this.data.getAction(event.action)
+
+		if (!action || !action.cooldown) { return }
+
+		const window = this.currentWindow
+		const observedUseTime = event.timestamp
+
+		// Cap at this event's timestamp, just in case
+		const earliestUseTime = Math.min(window.start + action.cooldown, observedUseTime)
+
+		let expectedUseTime = earliestUseTime
+
+		// Increase the expected use time if it was in downtime
+		if (this.downtime.isDowntime(earliestUseTime)) {
+			const downtimeWindow = this.downtime.getDowntimeWindows(window.start, observedUseTime)[0]
+			expectedUseTime = downtimeWindow.end
+
+			// Forgive "drift" due to reopening with other actions after downtime
+			if (downtimeWindow.end - downtimeWindow.start > MIN_REOPENER_DOWNTIME) {
+				expectedUseTime += REOPENER_BUFFER
+			}
+		}
+
+		const drift = Math.max(0, observedUseTime - expectedUseTime)
+
+		// Forgive a small amount of drift
+		if (drift > DRIFT_BUFFER) {
+			this.driftedWindows.push({
+				...window,
+				driftedActionId: event.action,
+				drift: drift,
+				end: observedUseTime,
+			})
+		}
+
+		// Begin the next window
+		this.currentWindow = {
+			start: observedUseTime,
+			casts: [],
+		}
+	}
+
+	private onCast(event: Events['action']) {
+		this.currentWindow.casts.push(event)
+	}
```

Comment:
It looks to me like we are not doing any analysis or display of the casts used in between EA casts, so can we remove this tracking and the associated hook for all player casts?

Reply:
Yep, I just kinda copied from MCH module and forgot to remove this part